appside/database/logger.js

```diff
@@ -0,0 +1,101 @@
+"use strict";
+
+/**
+ * For logging data to firebase
+ */
+class Logger {
+    /**
+     * Initialize the logger and open a connection to the cloud database
+     * 
+     * @requires - no other logger has been constructed (singleton)
+     */
+    constructor() {
+        // Unique but non-secret identifiers.
+        // For more info and to configure security, see:
+        //     https://firebase.google.com/docs/web/setup
+        //     https://firebase.google.com/docs/projects/learn-more
+        //     https://firebase.google.com/docs/database/security/quickstart
+        var firebaseConfig = {
+            apiKey: "AIzaSyBJB60vHefTp2wAH749dLLWFEW7vM3MtPE",
+            authDomain: "emotional-clarity.firebaseapp.com",
+            databaseURL: "https://emotional-clarity.firebaseio.com",
+            projectId: "emotional-clarity",
+            storageBucket: "emotional-clarity.appspot.com",
+            messagingSenderId: "650523646400",
+            appId: "1:650523646400:web:829a780d6fcc9351"
+        };
+
+        // Initialize Firebase
+        try {
+            let init = firebase.initializeApp(firebaseConfig);
+        } catch (err) {
+            console.error('Cannot initialize logger: firebase connection failed');
+            throw err;
+        }
+
+        let auth = firebase.auth();
+        // returns a Promise that will store the user credentials when resolved
+        this.signIn = auth.signInAnonymously();
+    }
+
+    /**
+     * Stores responses from obj into the database. Overwrites any values previously logged
+     * using this method in the current session. Keys and values of the input map will
+     * be cleaned of illegal characters and stored as strings.
+     *
+     * Data is stored in app-responses subtree, under current uid
+     *
+     * @param data - a map whose keys and values are primitives
+     */
+    logResponses(data) {
+        this.signIn.then(credential => {
+            for(let pair of data.entries()) {
+                let key = pair[0];
+                let value = pair[1];
```

Comment:
This structure currently makes sense.
Just wanted to know if this will be updated to store _3 primitive values_ in each entry to resemble our User Data object. If not, a slightly more specific description of what may be stored might be helpful.
_(Will user's self report/consent/feedback be stored here as well?)_